lib/GeoPointCaster.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 Mia s.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const through2 = require('through2')
+const lget = require('lodash.get')
+const lset = require('lodash.set')
+const { JSONPath } = require('jsonpath-plus')
+const { fieldsToSchema } = require('./GeoPointCaster.utils')
+
+class GeoPointCaster {
+  constructor(collectionDefinition) {
+    this._collectionSchema = collectionDefinition.schema ?? fieldsToSchema(collectionDefinition.fields)
+  }
+
+  asStream() {
+    return through2.obj((chunk, _, callback) => {
+      const castedChunk = this.castItem(chunk)
+      callback(null, castedChunk)
+    })
+  }
+
+  castItem(item) {
+    let response = item
+    const pointerSeparator = new RegExp('/', 'g')
+    const miaConfigurations = new RegExp('(properties|schema|items|__mia_configuration)', 'g')
+    const pathToCoordinates = JSONPath({
+      json: item,
+      path: '$..[?(@property === "coordinates")]',
+      resultType: 'pointer',
+    })
+      .map(pointer => pointer
+        .replace(pointerSeparator, '.')
+        .replace(miaConfigurations, '')
+        .split('.')
+        .filter(string => string.length > 0)
+        .join('.')
+      )
```

Comment:
On the "better the `replace` than the `split/filter/join`", I'd probably review this `map` fn.

```suggestion
      .map(pointer => pointer
        .replace(miaConfigurations, '')
        .replace(/\\+/g, '.')
      )
```

Or, in case there might be already some `.` here and there:
```suggestion
      .map(pointer => pointer
        .replace(miaConfigurations, '')
        .replace(/[\\\.]+/g, '.')
      )
```

(same as line 60, maybe you can wrap everything in a function and keep it DRY)